### PR TITLE
Time Synchronisation

### PR DESCRIPTION
**Pull request by DWicki**
Adding of Timesynchronisation as well as a generic fieldSkippEvaluator.

### CODE REVIEW

File: src/main/kotlin/ch/loewenfels/issuetrackingsync/custom/TimeSynchronizationFromJiraToRtcMapper.kt [hunk 1]
```diff
@@ -0,0 +1,52 @@
...
+        fieldname: String,
+        issue: Issue,
+        issueTrackingClient: IssueTrackingClient<in T>,
+        value: Any?
+    ) {
+        val splittedFieldName = fieldname.split(",")
+        when (splittedFieldName.size) {
+            2 -> {
+                val newEstimatedTime = value as Number
+                issueTrackingClient as IssueTrackingClient<IWorkItem>
```
Comment (schereradi) [inline src/main/kotlin/ch/loewenfels/issuetrackingsync/custom/TimeSynchronizationFromJiraToRtcMapper.kt]: Unchecked cast: IssueTrackingClient<in T> to IssueTrackingClient<IWorkItem>

File: src/main/kotlin/ch/loewenfels/issuetrackingsync/executor/fields/skipping/FieldSkippingEvaluatorFactory.kt [hunk 2]
```diff
@@ -0,0 +1,52 @@
...
+        evaluators[fieldMappingDefinition] = fieldMappingDefinition.fieldSkipEvalutors
+            .mapNotNull(this::createFieldSkippingEvaluator)
+            .toMutableList()
+        return evaluators[fieldMappingDefinition] ?: mutableListOf()
+    }
+
+    private fun createFieldSkippingEvaluator(fieldSkippingEvaluatorDefinition: FieldSkippingEvaluatorDefinition): FieldSkippingEvaluator? {
+        val clazz: Class<FieldSkippingEvaluator>?
+        try {
+            clazz = Class.forName(fieldSkippingEvaluatorDefinition.classname) as Class<FieldSkippingEvaluator>
```
Comment (schereradi) [inline src/main/kotlin/ch/loewenfels/issuetrackingsync/executor/fields/skipping/FieldSkippingEvaluatorFactory.kt]: Unchecked cast: Class<*>! to Class<FieldSkippingEvaluator>

File: src/main/kotlin/ch/loewenfels/issuetrackingsync/executor/fields/skipping/SkipFieldOnStatus.kt [hunk 3]
```diff
@@ -0,0 +1,23 @@
...
+
+open class SkipFieldOnStatus(fieldSkippingEvaluatorDefinition: FieldSkippingEvaluatorDefinition) :
+    FieldSkippingEvaluator(fieldSkippingEvaluatorDefinition) {
+    override fun <T> hasFieldToBeSkipped(
+        issueClient: IssueTrackingClient<in T>,
+        issueBuilder: Any,
+        issue: Issue,
+        fieldname: String
+    ): Boolean {
+        val propIssue = issue.proprietaryTargetInstance as? T ?: null
```
Comment (schereradi) [inline src/main/kotlin/ch/loewenfels/issuetrackingsync/executor/fields/skipping/SkipFieldOnStatus.kt]: Unchecked cast: Any? to T

Comment (schereradi) [inline src/main/kotlin/ch/loewenfels/issuetrackingsync/executor/fields/skipping/SkipFieldOnStatus.kt]: Right operand of elvis operator (?:) is useless if it is null

File: src/test/kotlin/ch/loewenfels/issuetrackingsync/custom/TimeSynchronizationFromJiraToRtcMapperTest.kt [hunk 4]
```diff
@@ -0,0 +1,54 @@
...
+    fun setValueRtc() {
+        // arrange
+        val testee = TimeSynchronizationFromJiraToRtcMapper()
+        val issue = TestObjects.buildIssue("MK-1")
+        issue.sourceUrl = "http://localhost/issues/MK-1"
+        val targetClient = Mockito.mock(RtcClient::class.java)
+        val plannedValue = 15
+        Mockito.`when`(targetClient.getTimeValueInMinutes(any(), any())).thenReturn(plannedValue)
+        val value = 45 as Number
+        val expectedValue = value
```
Comment (schereradi) [inline src/test/kotlin/ch/loewenfels/issuetrackingsync/custom/TimeSynchronizationFromJiraToRtcMapperTest.kt]: Variable 'expectedValue' is never used

File: src/test/kotlin/ch/loewenfels/issuetrackingsync/custom/TimeSynchronizationFromJiraToRtcMapperTest.kt [hunk 5]
```diff
@@ -0,0 +1,54 @@
...
+        val value = 0 as Number
+        val fieldName = "timeTracking"
+        // act
+        testee.setValue(issue, fieldName, issue, targetClient, value)
+        // assert
+        Mockito.verify(targetClient).setTimeValue(safeEq(issue), safeEq(issue), safeEq(fieldName), safeEq(value))
+    }
+
+    @Test
+    fun setValueRtc() {
```
Comment (schereradi) [inline src/test/kotlin/ch/loewenfels/issuetrackingsync/custom/TimeSynchronizationFromJiraToRtcMapperTest.kt]: Failing test...

Comment (DWicki) [reply src/test/kotlin/ch/loewenfels/issuetrackingsync/custom/TimeSynchronizationFromJiraToRtcMapperTest.kt]: Fixed

**Pull request merged**